src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
-\{JSON body} -> application/jose+json
-
-\{Non JSON body} -> application/jose
-
-|===
-
-=== Using JWE for Encryption of Request and Response
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST include the “Content-Type” HTTP header in all responses.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST set the “Cache-Control” HTTP header to no-store to prevent caching of sensitive information.
+====
+
+Cross-Origin Resource Sharing (CORS) is an HTTP header-based mechanism used to allow or disallow resource sharing
+between different origins (i.e., domains).
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services SHOULD allow only a specific domain or set of domains.
+
+Non-public RESTful services MUST NOT allow all domains (origins).
+====
+
+== HTTP message body
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All requests to RESTful services handling Category 1-2 data SHOULD be digitally signed by the API consumer using JWS.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All responses from RESTful services handling Category 1-2 data MUST be digitally signed by the API provider using JWS.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+To avoid obfuscating HTTP body data by wrapping the payload in a JWS envelope, the HTTP body MUST be represented as a
+detached JWS structure.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The signature MUST be provided in a custom “x-jws-signature” HTTP header.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The "application/jose" media type MUST be set to indicate that the content is signed.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All requests to RESTful services handling Category 3-4 data MUST be digitally signed by the API consumer using JWS
+(detached) and encrypted using JWE compact serialization.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All responses from RESTful services handling Category 3-4 data MUST be digitally signed by the API provided using JWS
+(detached) and encrypted using JWE compact serialization.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+All GET and DELETE requests MUST be converted to POST by appending the corresponding HTTP method to the URL (e.g., GET
+/employers/93017373 becomes POST /employers/93017373/get).
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The "application/jose+jwe" media type MUST be set to indicate that the content is encrypted.
+====
+
+== Injection
+
+Injection attacks, regardless of the technology they target, arise from the interpretation of untrusted data.
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+RESTful services MUST properly validate all input (length, type, format, range) and sanitize it (e.g., by coding or
+escaping) before interpretation.
+====
+
+== Logging, monitoring, and alerting
```

Comment:
This section (though interesting) exceeds the scope a bit because it touches quite a bit on internal logging systems that are also valid outside REST.
Related though, use of tracing IDs is specified in REST styleguide: https://www.gcloud.belgium.be/rest/#tracing

Reply:
That's true. Although it would hurt to reiterate that proper auditing is essential.

Reply:
Added a link to the tracing section of the REST guidelines.